azurelinuxagent/common/logger.py

```diff
@@ -77,11 +78,12 @@ def log(self, level, msg_format, *args):
             msg = msg_format
         time = datetime.now().strftime(u'%Y/%m/%d %H:%M:%S.%f')
         level_str = LogLevel.STRINGS[level]
+        thread_name = threading.currentThread().name
         if self.prefix is not None:
-            log_item = u"{0} {1} {2} {3}\n".format(time, level_str, self.prefix,
+            log_item = u"{0} {1} {2} {3} {4}\n".format(time, thread_name, level_str, self.prefix,
                                                    msg)
         else:
-            log_item = u"{0} {1} {2}\n".format(time, level_str, msg)
+            log_item = u"{0} {1} {2} {3}\n".format(time, thread_name, level_str, msg)
```

Comment:
Fyi, these changes will break some of the dungeon crawler tests, which expect a specific log format